#! /bin/sh /usr/share/dpatch/dpatch-run
## 04_gcc_4.3.dpatch by  <dktrkranz@ubuntu.com>
## DP: Fix FTBFS with Ubuntu toolchain

@DPATCH@
diff -urNad yap-5.1.1~/C/alloc.c yap-5.1.1/C/alloc.c
--- yap-5.1.1~/C/alloc.c	2006-03-24 18:15:18.000000000 +0100
+++ yap-5.1.1/C/alloc.c	2008-07-09 22:55:34.000000000 +0200
@@ -796,7 +796,7 @@
     itos(getpid(), &file[12]);
 #endif /* HAVE_TMPNAM */
 #endif /* HAVE_MKSTEMP */
-    fd = open(file, O_CREAT|O_RDWR);
+    fd = open(file, O_CREAT|O_RDWR, 0644);
     if (fd < 0) {
       Yap_Error(FATAL_ERROR, TermNil, "mmap could not open %s", file);
       return NULL;
@@ -909,7 +909,7 @@
     itos(getpid(), &file[12]);
 #endif /* HAVE_TMPNAM */
 #endif /* HAVE_MKSTEMP */
-    fd = open(file, O_CREAT|O_RDWR);
+    fd = open(file, O_CREAT|O_RDWR, 0644);
     if (fd < 0) {
       Yap_ErrorMessage = Yap_ErrorSay;
       snprintf4(Yap_ErrorMessage, MAX_ERROR_MSG_SIZE,
